scripts/commit_classifier.py

```diff
@@ -146,6 +155,23 @@ def __init__(self, model_name, repo_dir, git_repo_dir, method_defect_predictor_d
                 "8cc47f47ffb686a29324435a0151b5fabd37f865",
             )
 
+        if revision is not None:
+            assert phabricator_deployment is None
+            assert diff_id is None
+            self.revision = revision
+
+        if diff_id is not None:
+            assert phabricator_deployment is not None
+            assert revision is None
+
+        self.update_commit_db()
+
+        with hglib.open(self.repo_dir) as hg:
+            if phabricator_deployment is not None and diff_id is not None:
+                self.apply_phab(hg, phabricator_deployment, diff_id)
+
+                self.revision = hg.log(revrange="not public()")[0].node.decode("utf-8")
```

Comment:
You can move all this code in the `if diff_id is not None:` branch above

Reply:
`if diff_id is not None:` above`with hglib.open` ?

Reply:
But we will still check for the `phabricator_deployment` variable right ?